src/client/eppo-precomputed-client.ts

```diff
@@ -43,40 +46,56 @@ export type PrecomputedFlagsRequestParameters = {
   skipInitialPoll?: boolean;
 };
 
+export function convertContextAttributesToSubjectAttributes(
+  contextAttributes: ContextAttributes,
+): Attributes {
+  return {
+    ...(contextAttributes.numericAttributes || {}),
+    ...(contextAttributes.categoricalAttributes || {}),
+  };
+}
+
+interface EppoPrecomputedClientOptions {
+  precomputedFlagStore: IConfigurationStore<PrecomputedFlag>;
+}
+
 export default class EppoPrecomputedClient {
   private readonly queuedAssignmentEvents: IAssignmentEvent[] = [];
   private assignmentLogger?: IAssignmentLogger;
   private assignmentCache?: AssignmentCache;
   private requestPoller?: IPoller;
   private precomputedFlagsRequestParameters?: PrecomputedFlagsRequestParameters;
   private subjectKey?: string;
-  private subjectAttributes?: Attributes;
-
-  constructor(
-    private precomputedFlagStore: IConfigurationStore<PrecomputedFlag>,
-    private isObfuscated = false,
-  ) {}
+  private subjectAttributes?: ContextAttributes;
+  private precomputedFlagStore: IConfigurationStore<PrecomputedFlag>;
 
-  public setPrecomputedFlagsRequestParameters(
-    precomputedFlagsRequestParameters: PrecomputedFlagsRequestParameters,
+  public constructor(
+    options: EppoPrecomputedClientOptions,
+    precomputedFlagsRequestParameters?: PrecomputedFlagsRequestParameters,
   ) {
-    this.precomputedFlagsRequestParameters = precomputedFlagsRequestParameters;
+    this.precomputedFlagStore = options.precomputedFlagStore;
+    if (precomputedFlagsRequestParameters) {
+      this.setPrecomputedFlagsRequestParameters(precomputedFlagsRequestParameters);
+      this.setSubjectData(
+        precomputedFlagsRequestParameters.precompute.subjectKey,
+        precomputedFlagsRequestParameters.precompute.subjectAttributes,
+      );
+    }
   }
 
-  public setSubjectAndPrecomputedFlagsRequestParameters(
-    precomputedFlagsRequestParameters: PrecomputedFlagsRequestParameters,
-  ) {
-    this.setPrecomputedFlagsRequestParameters(precomputedFlagsRequestParameters);
-    this.subjectKey = precomputedFlagsRequestParameters.precompute.subjectKey;
-    this.subjectAttributes = precomputedFlagsRequestParameters.precompute.subjectAttributes;
+  private setPrecomputedFlagsRequestParameters(parameters: PrecomputedFlagsRequestParameters) {
+    this.precomputedFlagsRequestParameters = parameters;
   }
 
-  public setPrecomputedFlagStore(precomputedFlagStore: IConfigurationStore<PrecomputedFlag>) {
-    this.precomputedFlagStore = precomputedFlagStore;
+  private setSubjectData(subjectKey: string, subjectAttributes: ContextAttributes) {
+    this.subjectKey = subjectKey;
+    this.subjectAttributes = subjectAttributes;
   }
 
-  public setIsObfuscated(isObfuscated: boolean) {
-    this.isObfuscated = isObfuscated;
```

Comment:
before we added the ability to parse whether a configuration is obfuscated from the configuration, when users needed to bootstrap the SDK in offline mode they need to be able to set this. the canonical example is the JS browser client (which by default is in obfuscated mode) getting a configuration from a nodejs server (which is not obfuscated).

that being said, I know there are customers using this method. removing this public method will require a major version bump please.

overall this is a more elegant solution to receive the obfuscation status from the configuration - users will need to upgrade both their nodejs and js browser SDKs to make sure that metadata is included.